bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -448,14 +448,20 @@ class PlotCanvasView extends Renderer.View
         @canvas.get("canvas_height") != height)
       @canvas_view.set_dims([width, height], trigger=false)
 
-    @canvas_view.render(force_canvas)
-
     for k, v of @renderer_views
       if not @range_update_timestamp? or v.set_data_timestamp > @range_update_timestamp
         @update_dataranges()
         break
 
     ctx = @canvas_view.ctx
+    ctx.save()  # Save default state
+
+    # Set hidpi-transform
+    ratio = @canvas_view.render(force_canvas)
```

Comment:
Right, I had fixed that in the next commit (which eventually did not make it in this PR). I will clean this up / add a clear todo. Sorry for the little sloppiness here.